### PR TITLE
MSSQL: Edit default 'encrypt' setting from false to true

### LINKED ISSUE #36632: MSSQL: Set default 'encrypt' setting to true instead of false
**What would you like to be added**:
I'd like the default encryption setting for MSSQL data sources when adding by UI or API to be true instead of the current default of false.

**Why is this needed**:
By having an encrypted connection by default, the likelihood of unintentional security misconfigurations is reduced. The option of a non-true encryption setting should be explicitly set.

### PR DESCRIPTION
**Pull request by CloudPlatformer**
**What this PR does / why we need it**:
Sets default encryption setting for MSSQL data sources to be true instead of the current default of false.

By having an encrypted connection by default, the likelihood of unintentional security misconfigurations is reduced. The option of a non-true encryption setting should be explicitly set.

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/36632

### CODE REVIEW

File: docs/sources/datasources/mssql.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ To access data source settings, hover your mouse over the **Configuration** (gea
 | `Authentication` | Authentication mode. Either using SQL Server Authentication or Windows Authentication (single sign on for Windows users).             |
 | `User`           | Database user's login/username                                                                                                        |
 | `Password`       | Database user's password                                                                                                              |
-| `Encrypt`        | This option determines whether or to which extent a secure SSL TCP/IP connection will be negotiated with the server, default `false`. |
+| `Encrypt`        | This option determines whether or to which extent a secure SSL TCP/IP connection will be negotiated with the server, default `true`. |
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/mssql.md:26]:
```suggestion
| `Encrypt`        | This option determines if and to what extent a secure SSL TCP/IP connection will get negotiated with the server. The default value is`true`.  |
```

**Review by achatterjee-grafana — CHANGES_REQUESTED** [review]
Added doc review.

File: docs/sources/datasources/mssql.md (hunk 1 above)
Comment (jack-cloud-platform) [reply docs/sources/datasources/mssql.md]: I've added that suggestion @achatterjee-grafana, thanks.

**Review by achatterjee-grafana — APPROVED** [review]
Doc change looks good.

**Comment by achatterjee-grafana** [comment]
Do we know how to fix the CI check fail? Quite a few other PRs are failing as well.

**Comment by jack-cloud-platform** [comment]
> Do we know how to fix the CI check fail? Quite a few other PRs are failing as well.

@achatterjee-grafana Could the CI check be requeued please as the check passed before my latest doc update?

**Comment by achatterjee-grafana** [comment]
I re-ran the C check and it is still failing. Do you need to merge main to your local? It may be a fix was pushed for the failure.

**Comment by jack-cloud-platform** [comment]
> I re-ran the C check and it is still failing. Do you need to merge main to your local? It may be a fix was pushed for the failure.

@achatterjee-grafana  I've merged main in and that's fixed it.

**Comment by achatterjee-grafana** [comment]
@jack-cloud-platform, I added a backport label, feel free to take it off in case this change is for 8.1. Added label, as PR had no milestone. Thanks.

**Comment by ying-jeanne** [comment]
Trying to test things around, but i got this error on provisioned data source when test connection even encrypt changed to false. On main it seems working.

Screenshot 2021-07-29 at 10 14 04

**Review by ying-jeanne — CHANGES_REQUESTED** [review]
Trying to test things around, it is a bit weird. On the branch, with the provisioned mssql data source, even with encrypt= false, the connection test failed. It seems working on main.

image

**Pull request closed without merging**